### PR TITLE
Azure firewall integration

### PR DESCRIPTION
**Pull request by yehudaQ**
## Contributing to Cortex XSOAR Content
Make sure to register your contribution by filling the [contribution registration form](https://forms.gle/XDfxU4E61ZwEESSMA)

**The Pull Request will be reviewed only after the contribution registration form is filled.**

## Status
- [x] In Progress
- [x] Ready
- [ ] In Hold - (Reason for hold)

## Related Issues
fixes: link to the issue

## Description
A few sentences describing the overall goals of the pull request's commits.

## Screenshots
Paste here any images that will help the reviewer

## Minimum version of Cortex XSOAR
- [ ] 6.0.0
- [ ] 6.1.0
- [x] 6.2.0
- [ ] 6.5.0

## Does it break backward compatibility?
   - [ ] Yes
       - Further details:
   - [x] No

## Must have
- [x] Tests
- [x] Documentation

### CODE REVIEW

File: Packs/AzureFirewall/Integrations/AzureFirewall/AzureFirewall.yml [hunk 1]
```diff
@@ -0,0 +1,1507 @@
...
+      required: false
+      secret: false
+    deprecated: false
+    description: Attach policy to firewall. The policy and firewall have to belong
+      to the same tier.
+    execution: false
+    polling: true
+    name: azure-firewall-policy-attach
+    outputs:
+    - contextPath: AzureFirewall.Firewall.id
```
Comment (yuvalbenshalom) [inline Packs/AzureFirewall/Integrations/AzureFirewall/AzureFirewall.yml:641]: Note to self, check context outputs with @meital-aizen

File: Packs/AzureFirewall/Integrations/AzureFirewall/AzureFirewall.yml [hunk 2]
```diff
@@ -0,0 +1,1507 @@
...
+  dockerimage: demisto/crypto:1.0.0.27243
+  feed: false
+  isfetch: false
+  longRunning: false
+  longRunningPort: false
+  runonce: false
+  script: '-'
+  subtype: python3
+  type: python
+fromversion: 5.5.0
```
Comment (yuvalbenshalom) [inline Packs/AzureFirewall/Integrations/AzureFirewall/AzureFirewall.yml:1505]:
```suggestion
fromversion: 6.0.0
```

File: Packs/AzureFirewall/Integrations/AzureFirewall/AzureFirewall.py [hunk 3]
```diff
@@ -0,0 +1,2848 @@
...
+    complete_requests = False
+    total_response = {'value': []}
+    response = client.azure_firewall_list_request(resource=resource)
+
+    while not complete_requests:
+        total_response['value'].extend(response.get('value'))
+        if len(total_response['value']) >= end_offset or not response.get('nextLink'):
+            complete_requests = True
+        else:
+            response = client.azure_firewall_list_request(resource=resource, next_link=response.get('nextLink'))
```
Comment (yuvalbenshalom) [inline Packs/AzureFirewall/Integrations/AzureFirewall/AzureFirewall.py:625]: if the API has paging handling(next link) shouldn't we be able to pass the page six and number, or limit/skip to the api? this way we will not need fill out the pages, just get to the next link.

**Review by yuvalbenshalom — CHANGES_REQUESTED** [review]
Looks good.
see my comments.

File: Packs/AzureFirewall/Integrations/AzureFirewall/AzureFirewall.py (hunk 3 above)
Comment (yehudaQ) [reply Packs/AzureFirewall/Integrations/AzureFirewall/AzureFirewall.py:625]: As we agreed with @meital-aizen  in our previous integrations - The pagination arguments should be `page` and `limit` - and we have to handle the code according to the API limitations and behavior.

File: Packs/AzureFirewall/Integrations/AzureFirewall/AzureFirewall.yml (hunk 2 above)
Comment (yehudaQ) [reply Packs/AzureFirewall/Integrations/AzureFirewall/AzureFirewall.yml]: Updated to 6.2.0 since most of the commands are `scheduled` commands

**Review by yuvalbenshalom — APPROVED** [review]

**Comment by yuvalbenshalom** [comment]
@jrosenpanw this is ready for docs review.
@meital-aizen please assist if there will be questions form docs team next week.

**Comment by ShirleyDenkberg** [comment]
@yuvalbenshalom @meital-aizen I don't have permissions to edit the files.  I edited the files offline and sent them to both of you.  Please replace the current files with the files I sent.  In addition, there is no content in the Pack README file.  Please ask the contributor to add content to that file.

**Comment by ShirleyDenkberg** [comment]
@yuvalbenshalom @meital-aizen I don't have permissions to edit the files.  I edited the files offline and sent them to both of you.  Please replace the current files with the files I sent.  In addition, there is no content in the Pack README file.  Please ask the contributor to add content to that file.

**Comment by ShirleyDenkberg** [comment]
@yuvalbenshalom Doc review completed.

**Comment by OfekMQmasters** [comment]
> @OfekMQmasters please fix validations and we are good to go.

@yuvalbenshalom I faced the same error in FortiSIEM integration. I tried to fix it by merging master to the branch but it did not resolve it.

**Comment by bakatzir** [comment]
related: https://github.com/demisto/etc/issues/41102

**Pull request merged**